pkg/torcx/profile_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 CoreOS Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package torcx
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+// format of test-get-profile.json:
+//
+// {
+//   "kind": "profile-manifest-v0",
+//   "value": {
+//     "images": [
+//       {
+//         "name": "test-name",
+//         "reference": "test-reference"
+//       }
+//     ]
+//   }
+// }
+
+func TestGetProfile(t *testing.T) {
+	profilePath := "../../fixtures/test-get-profile.json"
+
+	if _, err := os.Stat(profilePath); err != nil {
+		t.Fatal(err)
+	}
+
+	manifest, err := getProfile(profilePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if manifest.Kind != "profile-manifest-v0" {
+		t.Fatalf("manifest.kind mismatch: %s", manifest.Kind)
+	}
+
+	if len(manifest.Value.Images) == 0 {
+		t.Fatal("empty array for images")
+	}
+
+	image0 := manifest.Value.Images[0]
+	if image0.Name != "test-name" {
+		t.Fatalf("image name mismatch: %s", image0.Name)
+	}
+	if image0.Reference != "test-reference" {
+		t.Fatalf("image reference mismatch: %s", image0.Reference)
+	}
+}
+
+func TestPutGetProfile(t *testing.T) {
+	profilePath := "../../fixtures/test-put-profile.json"
```

Comment:
I'm not sure if there is a reason to pick an hardcoded path here. Wouldn't it be better/easier to just rely on `ioutil.TempFile`?